npm/pkg/dataplane/policies/policymanager_linux.go

```diff
@@ -12,14 +12,34 @@ import (
 )
 
 const (
-	maxTryCount             = 1
-	unknownLineErrorPattern = "line (\\d+) failed" // TODO this could happen if syntax is off or AZURE-NPM-INGRESS doesn't exist for -A AZURE-NPM-INGRESS -j hash(NP1) ...
-	knownLineErrorPattern   = "Error occurred at line: (\\d+)"
-
-	chainSectionPrefix        = "chain" // TODO are sections necessary for error handling?
-	maxLengthForMatchSetSpecs = 6       // 5-6 elements depending on Included boolean
+	maxTryCount = 2
+	// the error message doesn't describe the error if this pattern is within the message
+	// this could happen if syntax is off or AZURE-NPM-INGRESS doesn't exist for -A AZURE-NPM-INGRESS -j hash(NP1) ...
+	unknownLineErrorPattern = "line (\\d+) failed"
+	// the error message describes the error if this pattern is within the message
+	knownLineErrorPattern = "Error occurred at line: (\\d+)"
+
+	chainSectionPrefix = "chain"
 )
 
+/*
+Error handling for iptables-restore:
+Currently we retry on any error and will make two tries max.
+The section IDs and line error patterns are pointless currently. Although we can eventually use them to skip a section with an error and salvage the rest of the file.
```

Comment:
nit: Can you reduce this oneline into two, its too long.

Reply:
addressing in followup PR